### PR TITLE
Updated the documentation for Go SDK

### PR DESCRIPTION
**Pull request by gkats**
_Do not merge but please do review_ :smile: 

Replaced the existing docs with content for the v2 version of the SDK.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ When you're ready to submit your contribution, you're going to create a [pull re
 
 If this is your first time, or you need a refresher on how to create a PR, you can check out this video:
 
-[How to Contribute to an Open Source Project on GitHub}](https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github)
+[How to Contribute to an Open Source Project on GitHub](https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github)
```
Comment (gkats) [inline CONTRIBUTING.md:45]: This change is unrelated, I just noticed a typo and bundled the fix in.

Please note that there are other changes in this file, all about trailing whitespace. 

I have my editor configured to trim trailing whitespace on save. I would suggest we all do the same to avoid these messy diffs. :wink:

File: docs/references/go/overview.mdx [hunk 2]
```diff
@@ -5,32 +5,112 @@ description: Learn how to integrate Go into your Clerk application.
...
+
+For details on how to use this module, see the [Go Documentation](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2).
+
+The library has a resource-based structure which follows the way the [Clerk Backend API](https://clerk.com/docs/reference/backend-api) resources are organized.
+Each API supports specific operations, like Create or List. While operations for each resource vary, a similar pattern is applied throughout the library.
+
+In order to start using API operations the library needs to be configured with your Clerk API secret key. Depending on your use case,
+there's two ways to use the library; with or without a client.
+
+For most use cases, the API without a client is a better choice.
```
Comment (LekoArts) [inline docs/references/go/overview.mdx:36]: Why?

File: docs/references/go/overview.mdx [hunk 3]
```diff
@@ -5,32 +5,112 @@ description: Learn how to integrate Go into your Clerk application.
...
+
+On the other hand, if you need to set up multiple Clerk API keys, using clients for API operations provides more flexibility.
```
Comment (LekoArts) [inline docs/references/go/overview.mdx:38]: Why would one need to setup multiple API keys?

File: docs/references/go/verifying-sessions.mdx [hunk 4]
```diff
@@ -3,100 +3,153 @@ title: Veryifing a session | Clerk Go SDK
...
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
 
+The two middleware functions are [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
+and [RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization).
+Their difference is that the `RequireHeaderAuthorization` calls `WithHeaderAuthorization` under the hood, but responds
+with HTTP 403 Forbidden if it fails to detect valid session claims.
+
+Let's see how the middleware functions can be used in action.
```
Comment (LekoArts) [inline docs/references/go/verifying-sessions.mdx:25]: This feels like a filler sentence. I'd say it can be removed and the context around the code blocks below stays the same.

**Review by LekoArts — CHANGES_REQUESTED** [review]
Thanks for writing this!

I gave it a light read and made some review comments :)

File: docs/references/go/overview.mdx (hunk 3 above)
Comment (gkats) [reply docs/references/go/overview.mdx:38]: I don't have a specific use-case in mind. Perhaps I can change the sentence to "if you need more control and flexibility over the library's configuration"?

Would that make more sense?

File: docs/references/go/verifying-sessions.mdx (hunk 4 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Removed.

File: docs/references/go/overview.mdx (hunk 2 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Added an explanation. Let me know if it makes sense, or we should provide more details. I guess only the example can showcase the differences.

File: docs/references/go/overview.mdx [hunk 5]
```diff
@@ -5,32 +5,112 @@ description: Learn how to integrate Go into your Clerk application.
 
 # Go
 
-Clerk's Go SDK is a thin wrapper over the Clerk Backend API. Add the following import statement:
+Clerk's [Go SDK](https://github.com/clerk/clerk-sdk-go) is a wrapper over the [Clerk Backend API](https://clerk.com/docs/reference/backend-api).
+
+## Installation
+
+If you are using Go Modules and have a `go.mod` file in your project's root, you can import clerk-sdk-go directly.
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/overview.mdx]: Sure, updated!

File: docs/references/go/other-examples.mdx [hunk 6]
```diff
@@ -5,30 +5,73 @@ description: Explore different examples of how to utilize Clerk with Go.
 
 # Examples
 
-Below is a code snippet showing what you can do from your backend with Clerk. Look at the [API reference](/docs/references/backend/overview) for a full list of requests you can make from your backend.
+Let's see an example that calls a few different [Clerk Backend API](/docs/references/backend/overview) operations using the Clerk Go SDK.
 
-## Update metadata
+In the snippet below, we will create an organization and update its slug. We'll fetch all organization
+memberships, loop through them to get the first one and get more details about the user who is the member of
+the organization.
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/other-examples.mdx]: Updated.

File: docs/references/go/other-examples.mdx [hunk 7]
```diff
@@ -5,30 +5,73 @@ description: Explore different examples of how to utilize Clerk with Go.
 
 # Examples
 
-Below is a code snippet showing what you can do from your backend with Clerk. Look at the [API reference](/docs/references/backend/overview) for a full list of requests you can make from your backend.
+Let's see an example that calls a few different [Clerk Backend API](/docs/references/backend/overview) operations using the Clerk Go SDK.
 
-## Update metadata
+In the snippet below, we will create an organization and update its slug. We'll fetch all organization
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/other-examples.mdx]: Thanks, updated!

File: docs/references/go/overview.mdx (hunk 3 above)
Comment (LekoArts) [reply docs/references/go/overview.mdx:38]: Maybe change the sentence to explain that you can instanciate multiple clients in order to handle use cases like wanting to change multiple sites (requiring multiple API keys)?

Comment (gkats) [reply docs/references/go/overview.mdx]: Updated. Thanks for the suggestion!

File: docs/references/go/other-examples.mdx [hunk 8]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
 
 # Examples
 
-Below is a code snippet showing what you can do from your backend with Clerk. Look at the [API reference](/docs/references/backend/overview) for a full list of requests you can make from your backend.
+Let's see an example that calls a few different [Clerk Backend API](/docs/references/backend/overview) operations using the Clerk Go SDK.
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:8]:
```suggestion
The following example demonstrates how to use the Clerk Go SDK to execute [Clerk Backend API](/docs/references/backend/overview) operations.
```

I wanted to make the language a little more formal, and drop the "Let's", as we should avoid using first-person pronouns in the docs (I know we use them throughout the docs but that will change over time)

File: docs/references/go/other-examples.mdx [hunk 9]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
 
-## Update metadata
+In the snippet below, you will create an organization and update its slug. You'll fetch all organization
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:10]:
```suggestion
By executing the code in the snippet below, you will:

- Create an organization and update its slug
- Fetch all organization memberships
```

File: docs/references/go/other-examples.mdx [hunk 10]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
+memberships, loop through them to get the first one, and get more details about the organization's user.
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:11]:
```suggestion
- Get more details about the organization's user
```

File: docs/references/go/other-examples.mdx [hunk 11]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
 
-Add a new key/value to the private metadata stored on a user.
+Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:13]:
```suggestion
<Callout type="info">

Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.

</Callout>
```

This feels appropriate for a callout to me

File: docs/references/go/other-examples.mdx [hunk 12]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
 
+<InjectKeys>
 ```go
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:16]: Is it possible to add a file path and filename here, even if placeholder, to make it explicit where this code might live in the user's codebase?

File: docs/references/go/other-examples.mdx [hunk 13]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
+  })
+  if err != nil {
+    // You can get additional information on the error, if it can
+    // be type-cast to clerk.APIErrorResponse.
+    if apiErr, ok := err.(*clerk.APIErrorResponse); ok {
+      apiErr.TraceID
+      apiErr.Error()
+      apiErr.Response.RawJSON
+    }
+    // handle the error
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:43]:
```suggestion
    // handle the error
    panic(err)
```

Does it make sense to add `panic` here as well?

File: docs/references/go/other-examples.mdx [hunk 14]
```diff
@@ -5,30 +5,72 @@ description: Explore different examples of how to utilize Clerk with Go.
...
+  // List organization memberships
+  listParams := organizationmembership.ListParams{}
+  listParams.Limit = clerk.Int64(10)
+  memberships, err := organizationmembership.List(ctx, params)
+  if err != nil {
+    // handle the error
+    panic(err)
+  }
+  if memberships.TotalCount < 0 {
+      return
```
Comment (S3Prototype) [inline docs/references/go/other-examples.mdx:64]: Should we log something into the console here so the user knows why the script did an early return in this case?

File: docs/references/go/overview.mdx [hunk 15]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
 
 # Go
 
-Clerk's Go SDK is a thin wrapper over the Clerk Backend API. Add the following import statement:
+Clerk's [Go SDK](https://github.com/clerk/clerk-sdk-go) is a wrapper over the [Clerk Backend API](https://clerk.com/docs/reference/backend-api).
+
+## Installation
+
+If you are using Go Modules and have a `go.mod` file in your project's root, you can import `clerk-sdk-go` directly.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:12]:
```suggestion
If you're using Go Modules and have a `go.mod` file in your project's root, you can import `clerk-sdk-go` directly.
```

File: docs/references/go/overview.mdx [hunk 16]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
 
 ```go
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:14]: Should this have the `go.mod` file path?

File: docs/references/go/overview.mdx [hunk 17]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
-import "github.com/clerkinc/clerk-sdk-go/clerk"
+import (
+    "github.com/clerk/clerk-sdk-go/v2"
+)
 ```
 
-Go doesn't automatically add the module. You can explicitly add it with:
+Alternatively, you can `go get` the package explicitly.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:20]:
```suggestion
Alternatively, you can `go get` the package explicitly:
```

File: docs/references/go/overview.mdx [hunk 18]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
 
-## `Clerk` client
+## Usage
+
+For details on how to use this module, see the [Go Documentation](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2).
+
+The library has a resource-based structure which follows the way the [Clerk Backend API](https://clerk.com/docs/reference/backend-api) resources are organized.
+Each API supports specific operations, like Create or List. While operations for each resource vary, a similar pattern is applied throughout the library.
+
+In order to start using API operations the library needs to be configured with your Clerk API secret key. Depending on your use case,
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:33]:
```suggestion
To execute API operations, you must configure the library with your Clerk API secret key. Depending on your use case,
```

**Comment by S3Prototype** [comment]
Looking good 👍 Will continue reviewing after lunch

File: docs/references/go/overview.mdx (hunk 16 above)
Comment (gkats) [reply docs/references/go/overview.mdx:14]: No, this import can (and will) be defined in every file you need to access the `clerk` package.

File: docs/references/go/other-examples.mdx (hunk 8 above)
Comment (gkats) [reply docs/references/go/other-examples.mdx]: Updated.

File: docs/references/go/other-examples.mdx (hunk 9 above)
Comment (gkats) [reply docs/references/go/other-examples.mdx]: Nice, updated!

File: docs/references/go/other-examples.mdx (hunk 11 above)
Comment (gkats) [reply docs/references/go/other-examples.mdx:16]: :+1: Updated.

File: docs/references/go/other-examples.mdx (hunk 13 above)
Comment (gkats) [reply docs/references/go/other-examples.mdx:47]: Sure, added!

File: docs/references/go/overview.mdx (hunk 15 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Done.

File: docs/references/go/overview.mdx [hunk 19]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+```sh filename="terminal"
+go get -u github.com/clerk/clerk-sdk-go/v2
 ```
 
-## `Clerk` client
+## Usage
+
+For details on how to use this module, see the [Go Documentation](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2).
+
+The library has a resource-based structure which follows the way the [Clerk Backend API](https://clerk.com/docs/reference/backend-api) resources are organized.
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/overview.mdx]: Updated.

File: docs/references/go/overview.mdx (hunk 18 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Updated.

File: docs/references/go/other-examples.mdx (hunk 14 above)
Comment (S3Prototype) [reply docs/references/go/other-examples.mdx:69]: Oh, okay! Yeah if the whole example doesn't do that, no need to add it here. Thanks for the explanation

File: docs/references/go/overview.mdx [hunk 20]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
-## `Clerk` client
+## Usage
+
+For details on how to use this module, see the [Go Documentation](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2).
+
+The library is organized using a resource-based structure similar to the [Clerk Backend API](https://clerk.com/docs/reference/backend-api).
+Each API supports specific operations, like Create or List. While operations for each resource vary, a similar pattern is applied throughout the library.
+
+To execute API operations, you must configure the library with your Clerk API secret key. Depending on your use case,
+there's two ways to use the library; with or without a client.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:34]:
```suggestion
there are two ways to use the library: With or without a client.
```

File: docs/references/go/overview.mdx [hunk 21]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+
+For most use cases, the API without a client is a better choice. It requires a minimal setup and supports a smaller API footprint.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:36]: What does "smaller API footprint" mean in this context?

File: docs/references/go/overview.mdx [hunk 22]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+
+However, if for some reason you need to operate on multiple Clerk instances from one application, or need more flexibility for tests and mocking,
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:38]:
```suggestion
However, if you need to operate on multiple Clerk instances from one application, or need more flexibility for tests and mocking,
```

File: docs/references/go/overview.mdx [hunk 23]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+you can instantiate multiple API clients with different API keys.
+
+Let's see both approaches in detail.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:41]:
```suggestion
The following examples demonstrate both approaches.
```

File: docs/references/go/overview.mdx [hunk 24]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+list, err := $resource$.List(ctx, &$resource$.ListParams{})
+for _, resource := range list.$Resource$s {
+    // do something with the resource
+}
 ```
 
-</InjectKeys>
+### Usage with a client
+
+If you're dealing with multiple keys, it is recommended to use a client based approach. The API packages for each
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:80]:
```suggestion
If you're working with multiple keys, you should use the library with a client. The API packages for each
```

File: docs/references/go/overview.mdx [hunk 25]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+resource export a Client, which supports all the API's operations.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:81]:
```suggestion
resource export a `Client`, which supports all the API's operations.
```

File: docs/references/go/overview.mdx [hunk 26]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+This way you can create as many clients as needed, each with their own API key.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:82]:
```suggestion
This way you can create as many clients as needed, each with their own API key, as shown in the following example:
```

File: docs/references/go/verifying-sessions.mdx [hunk 27]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
 description: Learn how to verify a session with Clerk in your Go application.
 ---
 
-# Verifying a session in Go
+# Verifying a Clerk session in Go
 
-## Protect your backend APIs
+Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples show how to verify a session and retrieve the corresponding user.
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:8]:
```suggestion
Go enables you to create a simple HTTP server, and Clerk enables you to authenticate any request. The following examples show how to verify a session and retrieve the corresponding user.
```

File: docs/references/go/verifying-sessions.mdx [hunk 28]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
 
-Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples shows how to verify a session and retrieve the corresponding user.
+## Using middleware
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:10]:
```suggestion
## Use middleware
```

File: docs/references/go/verifying-sessions.mdx [hunk 29]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
 
-<InjectKeys>
+The library provides two functions that can be used for adding authentication with Clerk to HTTP handlers.
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:12]:
```suggestion
The library provides two functions for adding authentication to HTTP handlers with Clerk.
```

File: docs/references/go/verifying-sessions.mdx [hunk 30]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
 
-```go
-package main
+Both middleware functions support header based authentication with a bearer token. The token is parsed, verified and
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:14]:
```suggestion
Both middleware functions support header based authentication with a bearer token. The token is parsed, verified, and
```

File: docs/references/go/verifying-sessions.mdx [hunk 31]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+its claims are extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:15]:
```suggestion
its claims will be extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
```

File: docs/references/go/verifying-sessions.mdx [hunk 32]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
 
+The two middleware functions are [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:20]:
```suggestion
There are also two middleware helpers:
- [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
```

File: docs/references/go/verifying-sessions.mdx [hunk 33]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+and [RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization).
+Their difference is that the `RequireHeaderAuthorization` calls `WithHeaderAuthorization` under the hood, but responds
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:22]:
```suggestion
```
Unnecessary due to above change

File: docs/references/go/verifying-sessions.mdx [hunk 34]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+with HTTP 403 Forbidden if it fails to detect valid session claims.
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:23]:
```suggestion
```
Unnecessary due to above change

File: docs/references/go/verifying-sessions.mdx [hunk 35]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+
+Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:25]:
```suggestion
<Callout type="info">

Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by selecting the eye icon below. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.

</Callout>
```

File: docs/references/go/verifying-sessions.mdx [hunk 36]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
+It is recommended to use the middleware functions for verifying Clerk sessions in an HTTP context.
 
-<InjectKeys>
+The middleware guarantees better performance and efficiency by making the minimum necessary requests to the Clerk Backend API.
 
-```go
-package main
+Verifying a Clerk session requires providing a JSON Web Key. The middleware fetches the key once and caches it.
+
+If you decide to verify the session token on your own, please be mindful of API rate limits. It is recommended to cache your JSON Web Key and invalidate the cache only when the key gets rolled.
```
Comment (S3Prototype) [inline docs/references/go/verifying-sessions.mdx:75]:
```suggestion
If you decide to verify the session token on your own, please be mindful of API rate limits. It is recommended to cache your JSON Web Key and invalidate the cache only when a replacement API key is generated.
```

Just wanted to make this slightly less technical, as I'd never heard of the term "Rolling" an API key until I looked it up just now. If you think that's a common enough term, you can ignore this suggestion, as I'll defer to your expertise. Just wanna make sure this content is accessible to all readers 👍

**Review by S3Prototype — APPROVED** [review]
This is awesome, thanks for contributing to the docs! Approved with a few suggested changes 👍

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexisintech) [reply CONTRIBUTING.md:45]: ack! just turned it on :)

File: docs/references/go/verifying-sessions.mdx (hunk 36 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Good thinking. Updated.

File: docs/references/go/verifying-sessions.mdx (hunk 35 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx:25]: Done!

File: docs/references/go/verifying-sessions.mdx (hunk 33 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: :+1: Yup, removed.

File: docs/references/go/verifying-sessions.mdx [hunk 37]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
-```go
-package main
+Both middleware functions support header based authentication with a bearer token. The token is parsed, verified and
+its claims are extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
 
+The two middleware functions are [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
+and [RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization).
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Done.

File: docs/references/go/verifying-sessions.mdx (hunk 32 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Switched to a list.

File: docs/references/go/verifying-sessions.mdx [hunk 38]
```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
...
-<InjectKeys>
+The library provides two functions that can be used for adding authentication with Clerk to HTTP handlers.
 
-```go
-package main
+Both middleware functions support header based authentication with a bearer token. The token is parsed, verified and
+its claims are extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Updated.

File: docs/references/go/verifying-sessions.mdx (hunk 31 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Done.

File: docs/references/go/verifying-sessions.mdx (hunk 30 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Done.

File: docs/references/go/verifying-sessions.mdx (hunk 29 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: Updated.

File: docs/references/go/verifying-sessions.mdx (hunk 27 above)
Comment (gkats) [reply docs/references/go/verifying-sessions.mdx]: You're 100% right. Updated!

File: docs/references/go/overview.mdx (hunk 26 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Updated.

File: docs/references/go/overview.mdx (hunk 25 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Done.

File: docs/references/go/overview.mdx (hunk 24 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Updated.

File: docs/references/go/overview.mdx (hunk 23 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Updated.

File: docs/references/go/overview.mdx (hunk 22 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Deleted.

File: docs/references/go/overview.mdx (hunk 21 above)
Comment (gkats) [reply docs/references/go/overview.mdx]: Reworded this a bit.

**Comment by alexisintech** [comment]
hey @gkats !
for the [verifying sessions](https://docs-preview-712.clerkpreview.com/docs/references/go/verifying-sessions) guide, are the two sections "Using middleware" and "Custom token verification" two different methods for verifying a session?
As in, a user can either use middleware to verify a session, _or_ they can manually verify a token/session ?

**Comment by gkats** [comment]
> As in, a user can either use middleware to verify a session, or they can manually verify a token/session ?

Hey @alexisintech, you are correct. The middleware functions are supposed to work in an HTTP server context.

We offer these middleware as helpers. They are using the `jwt.Verify` method internally. The `Verify` method is the primitive/core function here and the middleware is a convenience method around token verification in an HTTP server.

Most of the times, there's no need to reach out to `Verify` directly, but in case you want total control over the verification process, or you want to verify a token in a non-HTTP context, you can use `Verify` directly.

Hope that makes it clearer. I'm not sure if we should add this information to the docs, or if it's clear to the reader that 
- Have HTTP server?
  - Yes: Use the middleware
  - No: Use the custom flow

Let me know what you think!

**Comment by alexisintech** [comment]
> > As in, a user can either use middleware to verify a session, or they can manually verify a token/session ?
> 
> Hey @alexisintech, you are correct. The middleware functions are supposed to work in an HTTP server context.
> 
> We offer these middleware as helpers. They are using the `jwt.Verify` method internally. The `Verify` method is the primitive/core function here and the middleware is a convenience method around token verification in an HTTP server.
> 
> Most of the times, there's no need to reach out to `Verify` directly, but in case you want total control over the verification process, or you want to verify a token in a non-HTTP context, you can use `Verify` directly.
> 
> Hope that makes it clearer. I'm not sure if we should add this information to the docs, or if it's clear to the reader that
> 
> * Have HTTP server?
>   
>   * Yes: Use the middleware
>   * No: Use the custom flow
> 
> Let me know what you think!

Yes I do think we can make it more clear to users that we are offering two different solutions for verifying tokens! I have some edits in mind, so I reached out to you via pm :)

File: docs/references/go/overview.mdx [hunk 39]
```diff
@@ -5,32 +5,113 @@ description: Learn how to integrate Go into your Clerk application.
...
+For most use cases, the API without a client is a better choice. It requires a minimal setup and provides a more concise API for invoking operations.
+
+However, if you need to operate on multiple Clerk instances from one application, or need more flexibility for tests and mocking,
+you can instantiate multiple API clients with different API keys.
+
+The following examples demonstrate both approaches.
+
+### Usage without a client
+
+If you only use one API key, you can import the packages required for the APIs you need to interact with and call functions for API operations.
```
Comment (S3Prototype) [inline docs/references/go/overview.mdx:45]: I don't think I understand what this sentence means. @gkats If we reworded it this way, would it still be accurate?

> If you only use one API key, you can import the packages required for the APIs you need. Then you can execute your desired request methods as functions, such as `$resource$.Get()` or `$resource$.Delete()`. The following example demonstrates this process:

Comment (gkats) [reply docs/references/go/overview.mdx]: Yes! Updated.

File: docs/references/go/other-examples.mdx [hunk 40]
```diff
@@ -5,30 +5,77 @@ description: Explore different examples of how to utilize Clerk with Go.
 
 # Examples
```
[the remark this answers is not in the document]
Comment (gkats) [reply docs/references/go/other-examples.mdx]: Done!

**Comment by gkats** [comment]
> I'm so glad we could collaborate effectively on this !

Indeed! Thanks so much for your help @alexisintech and @S3Prototype. 

I'm planning to merge this into main next week. Hope it's ok to keep this open for a while.
